lightning/src/ln/msgs.rs

```diff
@@ -1515,6 +1586,170 @@ impl_writeable_len_match!(NodeAnnouncement, {
 	contents
 });
 
+impl Readable for QueryShortChannelIds {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let chain_hash: BlockHash = Readable::read(r)?;
+
+		let encoding_len: u16 = Readable::read(r)?;
+		if encoding_len == 0 || (encoding_len - 1) % 8 != 0 {
+			return Err(DecodeError::InvalidValue);
+		}
+
+		let encoding_type: u8 = Readable::read(r)?;
+		if encoding_type != EncodingType::Uncompressed as u8 {
```

Comment:
Maybe comment we expect 1-byte length `short_channel_id`. Also that queries/replies must be uncompressed.

Reply:
Do you mean 8-byte length `short_channel_id`?

Reply:
Ah sorry, you're right !

Reply:
No worries! Just wanted to make sure I wasn't misunderstanding how things worked haha

Reply:
Added comments about encoding_len, encoding_type, and reading the short_channel_ids